presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp

```diff
@@ -1380,6 +1380,19 @@ VeloxQueryPlanConverterBase::toVeloxQueryPlan(
       toVeloxQueryPlan(node->source, tableWriteInfo, taskId));
 }
 
+velox::VectorPtr VeloxQueryPlanConverterBase::evaluateConstantExpression(
+    const velox::core::TypedExprPtr& expression) {
+  auto emptyRowVector = BaseVector::create<RowVector>(ROW({}), 1, pool_);
+  core::ExecCtx execCtx{pool_, queryCtx_};
+  exec::ExprSet exprSet{{expression}, &execCtx};
+  exec::EvalCtx context(&execCtx, &exprSet, emptyRowVector.get());
+
+  SelectivityVector rows{1};
+  std::vector<VectorPtr> result(1);
```

Comment:
QQ : Would this vector need to be allocated from the memory pool if the result is a very large vector ?

Reply:
Yes. This vector is allocated from the memory pool even if result is small.

Reply:
@mbasmanova : This code seemed a little odd since it has result(1) without allocating the vector. On looking further I realized that eval allocates the vector internally. So then its natural that it got allocated in the pool.

Would it work to set result{nullptr} here and then add a comment that eval allocates the vector in the pool to make it more clear ?

Reply:
This code looks pretty standard to me. std::vector's constructor takes number of elements (1). Passing nullptr won't work.